typing/typecore.ml

```diff
@@ -2852,6 +2893,10 @@ let is_local_returning_expr e =
   in
   let rec loop e =
     match e.pexp_desc with
+    | Pexp_apply
+        ({ pexp_desc = Pexp_extension({txt = "extension.unique"}, PStr []) },
+         [Nolabel, exp]) ->
+      loop exp
```

Comment:
This is a pre-existing issue, but probably these cases should work for `ocaml.unique`, `unique`, `local` and `ocaml.local` too.